bittensor/_subtensor/subtensor_impl.py

```diff
@@ -467,61 +468,65 @@ def register (
-            with bittensor.__console__.status(":satellite: Registering...({}/10)".format(attempts)) as status:
-                # verify wallet not already registered
-                if (wallet.is_registered( self )):
-                    bittensor.__console__.print(":white_heavy_check_mark: [green]Registered[/green]")
-                    return True
+            with bittensor.__console__.status(":satellite: Registering...({}/{})".format(attempts,max_allowed_attempts)) as status:
+
+                # pow failed
+                if pow_result is None:
+                    # might be registered already
+                    if (wallet.is_registered( self )):
+                        bittensor.__console__.print(":white_heavy_check_mark: [green]Registered[/green]")
+                        return True
                     
-                with self.substrate as substrate:
-                    try:
-                        if pow_result is None:
-                            # might be registered already
-                            raise "Failed registration attempt"
-                        
-                        call = substrate.compose_call( 
-                            call_module='SubtensorModule',  
-                            call_function='register', 
-                            call_params={ 
-                                'block_number': pow_result['block_number'], 
-                                'nonce': pow_result['nonce'], 
-                                'work': bittensor.utils.hex_bytes_to_u8_list( pow_result['work'] ), 
-                                'hotkey': wallet.hotkey.ss58_address, 
-                                'coldkey': wallet.coldkeypub.ss58_address
-                            } 
-                        )
-                        extrinsic = substrate.create_signed_extrinsic( call = call, keypair = wallet.hotkey )
-                        response = substrate.submit_extrinsic( extrinsic, wait_for_inclusion=wait_for_inclusion, wait_for_finalization=wait_for_finalization )
-                        # We only wait here if we expect finalization.
-                        if not wait_for_finalization and not wait_for_inclusion:
-                            bittensor.__console__.print(":white_heavy_check_mark: [green]Sent[/green]")
-                            return True
-                        response.process_events()
-                        if not response.is_success:
-                            bittensor.__console__.print(":cross_mark: [red]Failed[/red]: error:{}".format(response.error_message))
-                            attempts += 1
-                            if attempts > max_allowed_attempts: 
-                                bittensor.__console__.print( "[red]No more attempts.[/red]" )
-                                return False
-                            else:
-                                status.update( ":satellite: Registering...({}/10)".format(attempts))
+                else:
```

Comment:
no need for `else` here if we are returning True if wallet is registered. Can just proceed.

Reply:
This else is actually for the pow, not for if wallet is registered. We skipped substrate call if pow fails and retry the pow instead. Let me add a comment to make it more clear